url78f6619d18_IntersectAustralia_rifcs_tgz-pJ8-org_ands_rifcs_example_SampleRIFCSBuildCollectionJ8/src/Right.java

```diff
@@ -0,0 +1,122 @@
+/*     */ package org.ands.rifcs.base;
+/*     */ 
+/*     */ import org.w3c.dom.Element;
+/*     */ import org.w3c.dom.Node;
+/*     */ 
+/*     */ public class Right extends RIFCSElement
+/*     */ {
+/*  31 */   protected RightsInfo rightsStatement = null;
+/*  32 */   protected RightsTypedInfo licence = null;
+/*  33 */   protected RightsTypedInfo accessRights = null;
+/*     */ 
+/*     */   protected Right(Node n)
+/*     */     throws RIFCSException
+/*     */   {
+/*  44 */     super(n, "rights");
+/*     */   }
+/*     */ 
+/*     */   public void setRightsStatement(RightsInfo rightsStatement)
+/*     */     throws RIFCSException
+/*     */   {
+/*  54 */     this.rightsStatement = rightsStatement;
+/*  55 */     getElement().appendChild(this.rightsStatement.getElement());
+/*     */   }
+/*     */ 
+/*     */   public void setRightsStatement(String value)
+/*     */     throws RIFCSException
+/*     */   {
+/*  65 */     setRightsStatement(value, null);
+/*     */   }
+/*     */ 
+/*     */   public void setRightsStatement(String value, String rightsUri)
+/*     */     throws RIFCSException
+/*     */   {
+/*  76 */     RightsInfo rightsStatement = new RightsInfo(newElement("rightsStatement"));
+/*  77 */     rightsStatement.setRightsUri(value);
+/*  78 */     if (rightsUri != null)
+/*  79 */       rightsStatement.setRightsUri(rightsUri);
+/*  80 */     setRightsStatement(rightsStatement);
+/*     */   }
+/*     */ 
+/*     */   public RightsInfo getRightsStatement()
+/*     */   {
+/*  89 */     return this.rightsStatement;
+/*     */   }
+/*     */ 
+/*     */   public void setLicence(RightsTypedInfo licence)
+/*     */     throws RIFCSException
+/*     */   {
+/*  98 */     this.licence = licence;
+/*  99 */     getElement().appendChild(this.licence.getElement());
+/*     */   }
+/*     */ 
+/*     */   public void setLicence(String value)
+/*     */     throws RIFCSException
+/*     */   {
+/* 109 */     setLicence(value, null);
+/*     */   }
+/*     */ 
+/*     */   public void setLicence(String value, String type)
+/*     */     throws RIFCSException
+/*     */   {
+/* 120 */     setLicence(value, type, null);
+/*     */   }
+/*     */ 
+/*     */   public void setLicence(String value, String rightsUri, String type)
+/*     */     throws RIFCSException
+/*     */   {
+/* 132 */     RightsTypedInfo licence = new RightsTypedInfo(newElement("licence"));
+/* 133 */     licence.setValue(value);
+/* 134 */     if (rightsUri != null)
+/* 135 */       licence.setRightsUri(rightsUri);
+/* 136 */     if (type != null)
+/* 137 */       licence.setType(type);
+/* 138 */     setLicence(licence);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 68:74, 104:110.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.